### ISSUE TITLE
Support multiple open 'pools'

### ISSUE THREAD

**Issue by robertkarl** [robertkarl/yabc#64]
Required for #62.

Currently the algorithm looks like this:
```
for each different coin:
  select all tx's that appear to contain that coin
  create an empty pool
  for each tx we selected:
    call process_one()
```

At the very least we need to make sure that we select the right tx's for each coin.

**Comment by robertkarl** [comment]
Consider making the pool a map from strings onto lists.

**Comment by robertkarl** [comment]
Fixed by #67 with the CoinPool class
